### PR TITLE
refactor: icons using ui lib svgs

### PR DESCRIPTION
**Pull request by fbwoolf**
> Try out Leather build ee0e6fe — [Extension build](https://github.com/leather-io/extension/actions/runs/10579088030), [Test report](https://leather-io.github.io/playwright-reports/refactor-icons-using-svgs), [Storybook](https://refactor-icons-using-svgs--65982789c7e2278518f189e7.chromatic.com), [Chromatic](https://www.chromatic.com/library?appId=65982789c7e2278518f189e7&branch=refactor-icons-using-svgs)

This PR will install the new UI lib package into the extension and replaces all existing icons to now use the exported shared svg icons.

*Hoping I caught all the icon changes here bc now the icons won't be using the `styled.svg` and those token props. I tested it pretty thoroughly, but let me know if you see any that are off in size.

### CODE REVIEW

File: package.json
Comment (fbwoolf) [inline package.json]: To update.

File: package.json
Comment (fbwoolf) [inline package.json]: To remove.

**Comment by pete-watters** [comment]
Nice work here @fbwoolf 👍 This code LGTM but when I am testing this branch in the extension I am getting a lot of build errors. 

I have checked out and built your mono-repo branch `feat/web-svg-icons` and am installing that here via `file` but it seems like that's not working correctly as no doubt you have installed the package and tested it. 

I have tried deleting `node_modules` etc. but I can't get it working. It seems that locally for me `dist_web` is not being updated properly and most of the files it's looking for are in `dist_all`.

I see this error:
```
ERROR in ./node_modules/.pnpm/@leather.io+ui@file+..+mono+packages+ui_@babel+core@7.25.2_@babel+preset-env@7.25.4_@babel+co_vphq6r4upqy7qcmveyqsyhn5ja/node_modules/@leather.io/ui/dist-web/web.js 2:0-69
Module not found: Error: Can't resolve './src/components/accordion/accordion.web' in '/Users/petew/src/leather/extension/node_modules/.pnpm/@leather.io+ui@file+..+mono+packages+ui_@babel+core@7.25.2_@babel+preset-env@7.25.4_@babel+co_vphq6r4upqy7qcmveyqsyhn5ja/node_modules/@leather.io/ui/dist-web'
 @ ./src/app/components/loading-spinner.tsx 7:0-41 9:153-160
 @ ./src/app/app.tsx 15:0-73 41:79-101 49:128-150
 @ ./src/app/index.tsx 13:0-28 23:58-61
``` 

In the `mono` PR - https://github.com/leather-io/mono/pull/388, if I check that out and:
- `cd packages/ui`
- `rm -rf node_modules dist-web dist-all dist-native`
- `pnpm i` 
- `pnpm build:all` / `pnpm build` / `pnpm build:web` 

It's still outputting most of the files to `dist-all/` and when we are looking for them in `dist-web/` they aren't available 
Screenshot 2024-08-23 at 10 42 14

I'm not sure what's happening and I am hoping it's an issue on my side (user error!). Maybe we can merge https://github.com/leather-io/mono/pull/388 and have the package published and check there?

**Comment by fbwoolf** [comment]
> 
> I'm not sure what's happening and I am hoping it's an issue on my side (user error!). Maybe we can merge [leather-io/mono#388](https://github.com/leather-io/mono/pull/388) and have the package published and check there?

Yeah, seems like your process is a bit off maybe? You don't need to cd into the ui package just to build it and run all those cmds. This is def working, and it appears you have a local setup issue. I would have simply just deleted the build folders (even manually), and run `pnpm build` from the root > `pnpm i` > `pnpm dev` ...and then installed in your local extension repo. If you can just review the code here, that would be great. I appreciate you trying to test it.

EDIT: I can even remove the `dist-all` build altogether now, it isn't needed. Also, if you are only seeing `dist-all` it would mean to me that you are getting a build error somewhere and the others aren't completing? Why not just run, `pnpm build` from the root?

**Comment by fbwoolf** [comment]
@pete-watters I removed the `dist-all` altogether and builds/installs fine in the extension for me. This is a scrreenshot of my `node_modules` in the ext, all the needed files are located in the `dist-web` folder...

Screenshot 2024-08-23 at 9 15 14 AM

**Comment by fbwoolf** [comment]
And, running the extension...

Screenshot 2024-08-23 at 9 18 15 AM

**Comment by pete-watters** [comment]
> > I'm not sure what's happening and I am hoping it's an issue on my side (user error!). Maybe we can merge [leather-io/mono#388](https://github.com/leather-io/mono/pull/388) and have the package published and check there?
> 
> Yeah, seems like your process is a bit off maybe? You don't need to cd into the ui package just to build it and run all those cmds. This is def working, and it appears you have a local setup issue. I would have simply just deleted the build folders (even manually), and run `pnpm build` from the root > `pnpm i` > `pnpm dev` ...and then installed in your local extension repo. If you can just review the code here, that would be great. I appreciate you trying to test it.
> 
> EDIT: I can even remove the `dist-all` build altogether now, it isn't needed. Also, if you are only seeing `dist-all` it would mean to me that you are getting a build error somewhere and the others aren't completing? Why not just run, `pnpm build` from the root?

Thanks @fbwoolf. I'll retest this soon, I trust you that it is working and I am making some mistake / doing something wrong. 

The code does LGTM. I will re-try the steps you mention above shortly

**Comment by fbwoolf** [comment]
I can just merge/release the mono repo changes, was just hoping to install here quickly so dev doesn't stay broken for people after the mono merge.

**Comment by pete-watters** [comment]
> @pete-watters I removed the `dist-all` altogether and builds/installs fine in the extension for me. This is a scrreenshot of my `node_modules` in the ext, all the needed files are located in the `dist-web` folder...
> 
> Screenshot 2024-08-23 at 9 15 14 AM

OK so I retried again:
- using branch `feat/web-svg-icons`
- manually deleted all build files in the `mono/packages/ui` - `rm -rf node_modules dist-web dist-all dist-native`
- did an install and build at mono root (`pnpm i`, `pnpm build`) 
- I got no build errors and can see:
    - no `dist-all` folder since your latest update 
    - `dist-web` was recreated but all it contains is `web.js` and `web.d.ts` - I don't see the other files copied over
    
 Maybe there is some permissions issue for me running the copy? 
 
 To get this PR merged anyway we need to install the published package. I see you merged https://github.com/leather-io/mono/pull/388 so once we get that package published let's update this PR with the real package and test again

**Comment by fbwoolf** [comment]
~~I made some changes in the ui pkg and hoping this is fixed now with the latest push which installs the latest ui pkg release.~~

EDIT: Appears the typecheck errors indicate it isn't finding some of the leather-styles, so looking into it.

**Comment by fbwoolf** [comment]
I see the problem here and why we were using `dist-all`, but wondering if there is a better way to set it up bc we don't really need `all` in the extension?
https://github.com/egoist/tsup/issues/564
https://github.com/egoist/tsup/pull/566/files
Link from issue to MetaMask: https://github.com/MetaMask/utils/pull/144/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519

I will fix UI package tom so it isn't broken.

EDIT: Fwiw, MetaMask appears to have already moved away from tsup.

**Comment by pete-watters** [comment]
> I see the problem here and why we were using `dist-all`, but wondering if there is a better way to set it up bc we don't really need `all` in the extension? [egoist/tsup#564](https://github.com/egoist/tsup/issues/564) https://github.com/egoist/tsup/pull/566/files Link from issue to MetaMask: https://github.com/MetaMask/utils/pull/144/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519
> 
> I will fix UI package tom so it isn't broken.
> 
> EDIT: Fwiw, MetaMask appears to have already moved away from tsup.

Thanks @fbwoolf 👍 glad you are trying to find the root issue here. It would be better to not have the `dist-all` but I did try doing that before and couldn't get it to work stably.

**Review by pete-watters — APPROVED** [review]
I have re-tested this PR and it's working for me now without any console errors serving the app or in the browser. I can see the `dist-web` folder populated and no `dist-all`. Great work! 👏  

I noticed one visual issue with the colour of the arrows on the activity icons not being correct:
Screenshot 2024-08-26 at 07 46 24

I am happy to approve and merge this though and then have that issue fixed in a smaller follow up later. 


To get it merged we need to solve the typecheck issue. `dist-all` was including some more map files and types I don't see now. Maybe thats the issue?

Screenshot 2024-08-26 at 10 41 25

**Pull request merged**
